package.json

```diff
@@ -88,6 +89,13 @@
     "testRunner": "jest-circus/runner",
     "useStderr": true
   },
+  "pkg": {
+    "assets": [
+      "./node_modules",
```

Comment:
I believe we need to configure `assets` only because we hack [`module.paths`](https://github.com/serverless/serverless/blob/650d55da741837ce8ea5ebe2ab957c9b71dcab21/lib/classes/PluginManager.js#L147).

It should not be that difficult to get rid of that `module.path` hacking, and we may consider to do so, if we want to reduce final package size (so it's not all `node_modules` that are taken)

Reply:
Yes, that's exactly the reason. Would love to get rid of the `module.paths` hack. Any idea how to do that?

Reply:
> Any idea how to do that?

- Do not tweak `module.paths`
- Resolve plugin paths in context of service with help of [`ncjsm/resolve/sync`](https://github.com/medikoo/ncjsm#resolvesyncdir-path) (or async way with [`ncjsm/resolve`](https://github.com/medikoo/ncjsm#resolvedir-path) if it's fine to switch to async)

Reply:
Still as I'm thinking, above may not make big difference here (as it doesn't expose plain requires to core plugins, and this is what we miss).

What we actually need to do, is to replace [Plugin.json](https://github.com/serverless/serverless/blob/02304f522a1d0f86fcf546634f8f78d0f5a712ae/lib/plugins/Plugins.json)

With dictionary of required plugins as:

```javascript
module.exports.plugins = [
  require("./config/config.js"),
  require("./create/create.js"),
  ....
]
```

And require it and support it in `PluginManager.js`.

That should expose all missing paths to `pkg`

Reply:
Closing this convo as #6814 should've resolved the issues discussed here.